### PR TITLE
Check creating task with advanced params with file from cloud data

### PR DESCRIPTION
**Pull request by novda**
### Motivation and context

Test for https://github.com/cvat-ai/cvat/pull/7969
### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have created a changelog fragment 
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

## Summary by CodeRabbit

- **Tests**
  - Added tests for data retrieval and frame information in the REST API.
  - Updated test logic to handle video file generation and removal.
  - Enhanced test utility function to support additional specifications for tasks.

### CODE REVIEW

**Comment by zhiltsov-max** [comment]
In the PR being tested, there were changes related to cloud data downloading optimizations. The problem was reported by the users. Could you add a test for regressions in this regard?

File: tests/python/rest_api/test_tasks.py [hunk 1]
```diff
@@ -1723,6 +1745,91 @@ def test_create_task_with_cloud_storage_and_check_data_sorting(
...
+            use_cache=False,
+            server_files=["test/video/video.avi"],
+            org=org,
+            spec=data_spec,
+            data_type="video",
+            video_frame_count=4000,
+        )
+        after_memory = process.memory_info().rss
+        print(before_memory, after_memory)
+        print(f"Memory usage: {after_memory - before_memory}")
```
Comment (zhiltsov-max) [inline tests/python/rest_api/test_tasks.py:1824]: I doubt the `print` calls check anything. Also, rss is not maximum memory used, so has no relation to the actual memory requirements of the operation being tested.

Comment (novda) [reply tests/python/rest_api/test_tasks.py]: Yeah, I forgot to remove the `prints`, fixed that. And changed the memory usage calculation.

File: tests/python/rest_api/test_tasks.py [hunk 2]
```diff
@@ -1723,6 +1745,90 @@ def test_create_task_with_cloud_storage_and_check_data_sorting(
...
+            cloud_storage=cloud_storage,
+            use_manifest=False,
+            use_cache=False,
+            server_files=["test/video/video.avi"],
+            org=org,
+            spec=data_spec,
+            data_type="video",
+            video_frame_count=4000,
+        )
+        mem_usage_after = memory_usage(-1, interval=0.1, timeout=1)
```
Comment (zhiltsov-max) [inline tests/python/rest_api/test_tasks.py:1821]: I'm not sure the returned number has any relation to the import worker process. The docs say that pid -1 is for the current process and the any subprocesses will be returned in a list in the call result. Also, the value of 8 doesn't tell anything about the frames fetched during the import.

**Review by zhiltsov-max — APPROVED** [review]

File: tests/python/rest_api/test_tasks.py (hunk 2 above)
Comment (zhiltsov-max) [reply tests/python/rest_api/test_tasks.py]: Canceled as per discussion

**Pull request merged**